ServerSide/yoty/Core/Managers/Bids/BidsManager.cs

```diff
@@ -0,0 +1,291 @@
+﻿// Copyright (c) YOTY Corporation and contributors. All rights reserved.
+
+namespace YOTY.Service.Core.Managers.Bids
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Runtime.CompilerServices;
+    using System.Threading.Tasks;
+    using YOTY.Service.Data;
+    using YOTY.Service.Data.Entities;
+    using YOTY.Service.WebApi.PublicDataSchemas;
+
+    public class BidsManager : IBidsManager
+    {
+        private const string BidNotFoundFailString = "Failed, Bid not found";
+        private static YotyContext _context = new YotyContext();
+
+        public async Task<Response<BuyerDTO>> AddBuyer(BidBuyerJoinRequest bidBuyerJoinRequest)
+        {
+            BidEntity bid = await _context.Bids.FindAsync(bidBuyerJoinRequest.bidId).ConfigureAwait(false);
+            if (bid == null)
+            {
+                return new Response<BuyerDTO>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = BidNotFoundFailString };
+            }
+            bid.CurrentParticipancies.Add(new ParticipancyEntity {
+                BidId = bidBuyerJoinRequest.bidId,
+                BuyerId = bidBuyerJoinRequest.buyerId,
+                NumOfUnits = bidBuyerJoinRequest.Items
+            });
+            bid.UnitsCounter += bidBuyerJoinRequest.Items;
+            using (var new_context = new YotyContext())
+            {
+                try
+                {
+                    new_context.Bids.Update(bid);
```

Comment:
what we are updating here ? just the bid counters ?

Reply:
i think we dont neet to bid units counter as a property of bids since we could get it as sum of CurrentParticipancies.NumOfUnits .... i think you can remove this property